server-alias using comma separated server name, observe different behaviour

**NGINX Ingress controller version**: 0.26.1

**Kubernetes version** (use `kubectl version`): 1.15.5

**Environment**:

- **Cloud provider or hardware configuration**: AWS
- **OS** (e.g. from /etc/os-release): Debian GNU/Linux 9 (stretch)
- **Kernel** (e.g. `uname -a`): 4.9.0-11-amd64
- **Install tools**: using kops
- **Others**:

**What happened**: If server alias already configured in ingress with comma separated and if trying to add new ingress using host from alias, the getting below exception, 
also is alias are with space separated then nothing happen only getting warning "Aliases already configured for server"

` I0202 19:53:02.936485       8 event.go:255] Event(v1.ObjectReference{Kind:"Ingress", Namespace:"default", Name:"ingress-1", UID:"10a93387-45c7-11ea-918e-029e82fc6cea", APIVersion:"networking.k8s.io/v1beta1", ResourceVersion:"63533747", FieldPath:""}): type: 'Normal' reason: 'UPDATE' Ingress default/ingress-1
W0202 19:53:05.517811       8 controller.go:1089] Aliases already configured for server "www.test.com", skipping (Ingress "default/ingress-2")
W0202 19:53:05.517839       8 controller.go:1155] Conflicting hostname (www.test.com) and alias (test1.test.com). Removing alias to avoid conflicts.
W0202 19:53:05.517847       8 controller.go:1155] Conflicting hostname (www.test.com) and alias (www.test.com). Removing alias to avoid conflicts.
W0202 19:53:05.517854       8 controller.go:1155] Conflicting hostname (www.test.com) and alias (www.test.com). Removing alias to avoid conflicts.
E0202 19:53:05.517905       8 runtime.go:78] Observed a panic: runtime.boundsError{x:3, y:1, signed:true, code:0x3} (runtime error: slice bounds out of range [3:1])
goroutine 195 [running]:
k8s.io/ingress-nginx/vendor/k8s.io/apimachinery/pkg/util/runtime.logPanic(0x16aede0, 0xc001617840)
	/tmp/go/src/k8s.io/ingress-nginx/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:74 +0xa3
k8s.io/ingress-nginx/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/tmp/go/src/k8s.io/ingress-nginx/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:48 +0x82
panic(0x16aede0, 0xc001617840)
	/home/ubuntu/.gimme/versions/go1.13.1.linux.amd64/src/runtime/panic.go:679 +0x1b2
k8s.io/ingress-nginx/internal/ingress/controller.(*NGINXController).createServers(0xc0004501c0, 0xc0008671a0, 0x3, 0x4, 0xc00198fa70, 0xc0019856c0, 0x179d1e7)
	/tmp/go/src/k8s.io/ingress-nginx/internal/ingress/controller/controller.go:1156 +0x2580
k8s.io/ingress-nginx/internal/ingress/controller.(*NGINXController).getBackendServers(0xc0004501c0, 0xc0008671a0, 0x3, 0x4, 0x0, 0x0, 0xc0008e1d58, 0x3, 0x78, 0xb4)
	/tmp/go/src/k8s.io/ingress-nginx/internal/ingress/controller/controller.go:454 +0xe4
k8s.io/ingress-nginx/internal/ingress/controller.(*NGINXController).getConfiguration(0xc0004501c0, 0xc0008671a0, 0x3, 0x4, 0x4, 0xc0008669a0, 0x3, 0x3, 0xc000382be0)
	/tmp/go/src/k8s.io/ingress-nginx/internal/ingress/controller/controller.go:402 +0x80
k8s.io/ingress-nginx/internal/ingress/controller.(*NGINXController).syncIngress(0xc0004501c0, 0x164ca80, 0xc000421460, 0xc00b00ded6, 0x294cd84d10f7b)
	/tmp/go/src/k8s.io/ingress-nginx/internal/ingress/controller/controller.go:122 +0xcc
k8s.io/ingress-nginx/internal/task.(*Queue).worker(0xc0003e9890)
	/tmp/go/src/k8s.io/ingress-nginx/internal/task/queue.go:129 +0x30c
k8s.io/ingress-nginx/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc000614fa8)
	/tmp/go/src/k8s.io/ingress-nginx/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:152 +0x5e
k8s.io/ingress-nginx/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000987fa8, 0x3b9aca00, 0x0, 0x1852401, 0xc0000ac540)
	/tmp/go/src/k8s.io/ingress-nginx/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:153 +0xf8
k8s.io/ingress-nginx/vendor/k8s.io/apimachinery/pkg/util/wait.Until(...)
	/tmp/go/src/k8s.io/ingress-nginx/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88
k8s.io/ingress-nginx/internal/task.(*Queue).Run(0xc0003e9890, 0x3b9aca00, 0xc0000ac540)
	/tmp/go/src/k8s.io/ingress-nginx/internal/task/queue.go:61 +0x6b
created by k8s.io/ingress-nginx/internal/ingress/controller.(*NGINXController).Start
	/tmp/go/src/k8s.io/ingress-nginx/internal/ingress/controller/nginx.go:310 +0x48e
panic: runtime error: slice bounds out of range [3:1] [recovered]
	panic: runtime error: slice bounds out of range [3:1]
`

**What you expected to happen**: behaviour should be same if using comma or space


**How to reproduce it**: use below ingress config and just apply it and it will throw error, also if you remove comma and use only space it work.

`---
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  name: ingress-1
  annotations:
    nginx.ingress.kubernetes.io/server-alias: "test1.test.com, test2.test.com, www.test.com"
    nginx.ingress.kubernetes.io/configuration-snippet: |
      set $test "1";
spec:
  rules:
  - host: www.test.com
    http:
      paths:
      - path: /
        backend:
          serviceName: test-service
          servicePort: 80
`
`apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  name: ingress-2
spec:
  rules:
  - host: test1.test.com
    http:
      paths:
      - path: /test
        backend:
          serviceName: test-service
          servicePort: 80`

--- a/internal/ingress/controller/controller.go
+++ b/internal/ingress/controller/controller.go
@@ -1154,12 +1154,28 @@ func (n *NGINXController) createServers(data []*ingress.Ingress,
 	}
 
 	for host, hostAliases := range allAliases {
-		for index, alias := range hostAliases {
-			if _, ok := servers[alias]; ok {
-				klog.Warningf("Conflicting hostname (%v) and alias (%v). Removing alias to avoid conflicts.", host, alias)
-				servers[host].Aliases = append(servers[host].Aliases[:index], servers[host].Aliases[index+1:]...)
-			}
+		if _, ok := servers[host]; !ok {
+			continue
 		}
+
+		uniqAliases := sets.NewString()
+		for _, alias := range hostAliases {
+			if alias == host {
+				continue
+			}
+
+			if _, ok := servers[alias]; ok {
+				continue
+			}
+
+			if uniqAliases.Has(alias) {
+				continue
+			}
+
+			uniqAliases.Insert(alias)
+		}
+
+		servers[host].Aliases = uniqAliases.List()
 	}
 
 	return servers

